### ISSUE TITLE
Imprecision in XRA

### ISSUE THREAD

**Issue by ThomasHaas** [hernanponcedeleon/Dat3M#622]
I noticed that on EBR, XRA under `imm.cat` produces a smaller may-set for `rf` than under other memory models.
After some debugging, I found the source of this discrepancy. After reducing it to its core, I found this
```
acyclic (po | rf | fr) // Original
// ---- The following all give smaller rf-sets but have the same semantics
acyclic ((po | rf)+ | fr) 
acyclic ((po | rf | fr)+)
acyclic ((po | fr)+ | rf)
```
There are only two conclusions: Either our XRA handling of acyclicity axioms is imprecise ~or propagation through transitive closures is somehow wrong.~

EDIT: After quickly checking the code, I found that XRA for acyclicity axioms is indeed imprecise because it avoids the computation of the transitive must-closure. Now that we have must rf-edges between threads, it might be worth to compute the transitive closure.

**Comment by hernanponcedeleon** [comment]
I don't get the EDIT part of the comment. Where in the code we should compute the transitive closure?

**Comment by ThomasHaas** [comment]
In the `Acyclicity` axiom when doing top-down propagation.

**Comment by hernanponcedeleon** [comment]
method `Acyclicity.transitivelyDerivableMustEdges()`?

**Comment by ThomasHaas** [comment]
No, that one is correct. I believe but it only is called when doing the active set computation.
The issue is in both the `knowledgeClosure` functions, which use the must-set of the constrained relations without transitively closing it.

**Comment by hernanponcedeleon** [comment]
Is this still an issue after some of the recent changes, e.g., #1044?

I tried to reproduce the claim in the first msg but `imm.cat` does not really have an `acyclic (po | rf | fr) // Original` axiom.

**Comment by ThomasHaas** [comment]
The `acyclic (po | rf | fr) // Original` part was a reduced version ("After reducing it to its core, I found this ...").
I think you can take any memory model with an acyclicity axiom `acyclic (r)` and check if `acyclic(r^+)` yields better results.
Or alternatively, compare the results under imm.cat vs. simpler models.

From some quick testing, I couldn't see any differences anymore. So the issue might be gone.
